src/DocumentConstructionObserver.js

```diff
@@ -0,0 +1,61 @@
+import CustomElementInternals from './CustomElementInternals';
+
+export default class DocumentConstructionObserver {
+  constructor(internals, doc) {
+    /**
+     * @type {!CustomElementInternals}
+     */
+    this._internals = internals;
+
+    /**
+     * @type {!Document}
+     */
+    this._document = doc;
+
+    /**
+     * @type {MutationObserver|undefined}
+     */
+    this._observer = undefined;
+
+
+    // Simulate tree construction for all currently accessible nodes in the
+    // document.
+    this._internals.patchTree(this._document);
+    this._internals.upgradeTree(this._document);
```

Comment:
This should be fixed now, `upgradeTree` now attempts to patch elements before upgrading.